#24-Singly-linked-lists-and.md

```diff
@@ -0,0 +1,24 @@
+# Introduction to Singly Linked Lists and Doubly Linked Lists
+Singly linked lists and doubly linked lists are special types of data structure because they introduce the concept of pointers (links) and nodes. A linked list is a collection of similar items called nodes that seek to establish a sequence / order to the collection. There are two types of node -- the head and tail nodes and the head is always the first node.  Singly linked lists’ nodes can only access the next node, while doubly linked lists’ nodes can access both the previous and next node. The use of the term "singly" or "doubly" represent the number "links" that each node can establish with another node. Think of a line of lawn mowers strung together by rope.
+
+### Singly Linked Lists
+Singly linked lists’ nodes are connected only to the next node. Singly linked lists are great when you want to access data in the forward direction only. The insertion and deletion of nodes of nodes is easy. Singly linked list require less memory than doubly linked lists. However, the main problem with singly linked lists is that directly accessing the previous node is impossible. You experience examples of singly linked list when visiting a bank because the teller is typically required to process the customers in the order they arrive in. The problem with singly linked list becomes obvious when told to rewind a song but only the fast forward "button" works. 
+
+### Doubly Linked Lists
+In contrast, doubly linked lists’ nodes are connected to both the previous and next node and this difference is very important. The main advantage of a doubly linked list over a singly linked list is its ability to traverse easily both in a forward and backward directions. This means you now have the ability ot rewind the song to the beginnging. But doubly linked lists can be disadvantageous compared to singly linked lists as well. With regards to programming, extra space is needed to store the extra pointer in a doubly linked list. Additionally, the extra pointer can be burdensome, as all operations such as insertion would require more logic to support the need to point to both the next and previous nodes. 
+
+## Conclusion
+Singly linked lists are implemented in stacks and queues, which are the foundation of computer science. A bank teller's line is a perfect everyday example of its advantage; and a broken rewind button its disadvantage. The advantaged of doubly linked lists can be demonstrated when having a song that you want to fast forward or rewind -- a feature that singly linked list would not possess. Doubly linked lists are also used to undo operations because it involves a user’s history and can be used to track or determine the executions of multiple processes.  Similar to the doubly linked list, singly linked lists can also be used to undo functions but you would never be able to "redo"  or "undo the undo". 
```

Comment:
- replace second sentence with something like "They're great for moving forward, but not so much moving back, like a stereo with a broken rewind button."
- delete d in "advantaged"
- replace third sentence "A doubly linked list, in contrast, is a stereo with functioning rewind and fast forward buttons."
- I would move the last two sentences under the Doubly Linked List section; this isn't information that should be introduced in the conclusion.
- In sentence beginning "Doubly linked lists..." insert "and redo actions" after undo and delete "because it involves a user’s history"
- delete "Similar to the doubly linked list," and capitalize "singly"
- comma after "functions"
- change "would never" to "wouldn't"